components/chef-ui-library/src/templates/chef-page/chef-page.tsx

```diff
@@ -0,0 +1,189 @@
+ *  cancel-btn-text="Cancel"
+ * >
+ *  <chef-table>
+ *    <chef-thead>
+ *       <chef-tr>
+ *         <chef-th class="checkbox-row"></chef-th>
+ *         <chef-th>ID</chef-th>
+ *         <chef-th>Type</chef-th>
+ *       </chef-tr>
+ *    </chef-thead>
+ *    <chef-tbody>
+ *      <chef-tr >
+ *        <chef-td>
+ *         <chef-checkbox></chef-checkbox>
+ *        </chef-td>
+ *        <chef-td>1</chef-td>
+ *        <chef-td>apple</chef-td>
+ *      </chef-tr>
+ *    </chef-tbody>
+ *  </chef-table>
+ * </chef-page>
+ */
+@Component({
+  tag: 'chef-page',
+  styleUrl: './chef-page.scss'
+})
+export class ChefPage {
+
+  /**
+   * Text for page heading
+   */
+  @Prop({ reflectToAttr: true }) heading: string;
+
+  /**
+   * Text for page subheading
+   */
+  @Prop({ reflectToAttr: true }) subheading: string;
+
+  /**
+   * Text for error message
+   */
+  @Prop({ reflectToAttr: true }) errorMessage: string;
+
+  /**
+   * Text for secondary button
+   */
+  @Prop({ reflectToAttr: true }) secondaryBtnText: string;
+
+  /**
+   * Text for confirm button
+   */
+  @Prop({ reflectToAttr: true }) confirmBtnText = 'Confirm';
+
+  /**
+   * Text for cancel button
+   */
+  @Prop({ reflectToAttr: true }) cancelBtnText = 'Cancel';
+
+  /**
+   * Indicate confirm is loading
+   */
+  @Prop({ reflectToAttr: true }) confirmLoading = false;
+
+  /**
+   * Indicate page is loading
+   */
+  @Prop({ reflectToAttr: true }) pageLoading = false;
+
+  /**
+   * Disable confirm buttom
+   */
+  @Prop({ reflectToAttr: true }) disableConfirm = true;
+
+  @Event() secondaryConfirm: EventEmitter;
+  @Event() confirm: EventEmitter;
+  @Event() close: EventEmitter;
+
+  render() {
+    if (this.pageLoading) {
+      return [
+        <chef-loading-spinner size="50" fixed></chef-loading-spinner>
+      ];
+    } else {
+      return [
+        <div id="page-header" class="flex">
+
+          <div id="header-description" class="flex-left">
+            <chef-page-header>
+              <chef-heading>{this.heading}</chef-heading>
+              <chef-subheading>{this.subheading}</chef-subheading>
+            </chef-page-header>
+          </div>
+
+          <div id="close-button" class="flex-right">
+            <chef-button
+              secondary
+              class="close-button"
+              aria-label="Close"
+              onClick={this.handleClose.bind(this)}>
+              <chef-icon aria-hidden="true">close</chef-icon>
+            </chef-button>
+          </div>
+
+        </div>,
+
+        <div id="page-container">
+          <slot />
+        </div>,
+
+        <div id="page-footer">
+
+          {this.secondaryBtnText
+            ? <chef-button
+                secondary
+                onClick={this.handleSecondaryConfirm.bind(this)}>
+                {this.secondaryBtnText}
+              </chef-button>
+            : ''
+          }
+
+          <div id="right-buttons">
+
+            {(this.disableConfirm || this.confirmLoading)
```

Comment:
Not clear what purpose disableConfirm does here... seems to be related to my other comment on "Adding Member...".